app/services/survey.js

```diff
@@ -0,0 +1,164 @@
+import { registerDestructor } from '@ember/destroyable';
+import { tracked } from '@glimmer/tracking';
+import { TOAST_OPTIONS } from '../constants/toast-options';
+import { APPS } from '../constants/urls';
+import {
+  ANSWER_STATUS,
+  API_METHOD,
+  PATCH_API_CONFIGS,
+  ROLES,
+} from '../constants/live';
+import { generateWordCloud } from '../d3/word-cloud';
+
+export default class SurveyService extends Service {
+  @service router;
+  @service toast;
+  @service live;
+  @service fastboot;
+  @tracked answers = [];
+  @tracked approvedAnswers = [];
+  @tracked recentQuestion;
+  @tracked screenWidth;
+
+  constructor() {
+    super(...arguments);
+
+    const onResize = () => {
+      this.screenWidth = window.innerWidth;
+      this.showWordCloud();
+    };
+
+    if (!this.fastboot.isFastBoot) {
+      this.screenWidth = window.innerWidth;
+      window.addEventListener('resize', onResize);
+      registerDestructor(this, () => {
+        window.removeEventListener('resize', onResize);
+      });
+    }
+  }
+  setApprovedAnswers(approvedAnswers) {
+    this.approvedAnswers = approvedAnswers;
+  }
+
+  setAnswers(answers) {
+    this.answers = answers;
+  }
+
+  setRecentQuestion(question) {
+    this.recentQuestion = question;
+  }
+
+  async answerSubmitHandler(payload) {
+    let error = null,
+      answer = null;
+
+    try {
+      const answerResponse = await fetch(`${APPS.API_BACKEND}/answers`, {
+        method: API_METHOD.POST,
+        headers: {
+          'Content-Type': 'application/json',
+        },
+
+        body: JSON.stringify(payload),
+      });
+      answer = await answerResponse.json();
+
+      if (!answerResponse.ok)
+        return this.toast.error(answer.message, answer.error, TOAST_OPTIONS);
+
+      this.toast.success(answer.message, answer.error, TOAST_OPTIONS);
+
+      return { error: error, result: answer };
+    } catch (error) {
+      console.error('Error while submitting answer: ', error);
+      return { error: error, result: null };
+    }
+  }
+  async answerApproveHandler(id) {
+    const approvalPayload = {
+      status: ANSWER_STATUS.APPROVED,
+    };
+
+    try {
+      const approveResponse = await fetch(`${APPS.API_BACKEND}/answers/${id}`, {
+        ...PATCH_API_CONFIGS,
+        body: JSON.stringify(approvalPayload),
+      });
+
+      if (!approveResponse.ok) throw new Error();
+
+      this.toast.success(
+        'Answer approved successfully',
+        'Success',
+        TOAST_OPTIONS,
+      );
+    } catch (error) {
+      console.error('Error while approving answer: ', error);
+      this.toast.error('Error while approving answer', 'Error', TOAST_OPTIONS);
+    }
+  }
+
+  async answerRejectHandler(id) {
+    const rejectionPayload = {
+      status: ANSWER_STATUS.REJECTED,
+    };
+    try {
+      const rejectResponse = await fetch(`${APPS.API_BACKEND}/answers/${id}`, {
+        ...PATCH_API_CONFIGS,
+        body: JSON.stringify(rejectionPayload),
```

Comment:
Why we are not using ember data?

Reply:
The apis were slight different so haven't used ember data

Reply:
Can you tell  why we avoid ember-data in this case?

Reply:
Because all things are happening in realtime we don't need functionality like caching that's why.